### PR TITLE
Handle AllAlpha and AllBeta in SetFeatureGateDuringTest

### LINKED ISSUE #116151: unstable map iteration order causes test to sometimes leave non-GA features disabled
### What happened?

PR https://github.com/kubernetes/kubernetes/pull/115324 was reverted by #116062 because it (erroneously) relied on OpenAPIV3 feature to be enabled, but `TestPodSecurityGAOnly` would non-deterministically leave the feature disabled after running.

I tracked it down to the fact that this test disables all `PreRelease` or `Beta` features in a loop which iterates over a `map`, which are known to have unstable iteration order:
https://github.com/kubernetes/kubernetes/blob/e55f2a9b54e0f6bd9ed12e50752d7d2545ab0cab/test/integration/auth/podsecurity_test.go#L74-L81

When key `AllBeta` is processed before key `OpenAPIV3`, then `AllBeta` would disable `OpenAPIV3` before `OpenAPIV3`s default value is to be saved.

Normally, `AllBeta` ignores features that have been explicitly set, so it only affects those which fall back on the `Default` value. But this issue does not just affect `OpenAPIV3` but any non-GA feature.

I suggest the loop in `TestPodSecurityGAOnly` be made to skip the special feature keys such as `AllBeta`.

/cc @liggitt 
/cc @apel

### What did you expect to happen?

The value before every feature gate should be restored after running `TestPodSecurityGAOnly`.

### How can we reproduce it (as minimally and precisely as possible)?

This test always fails on my machine (just pasted it into `test/integration/auth/podsecurity_test.go`):

```go
func TestRestoreFeatures(t *testing.T) {
	var targetFeature featuregate.Feature

	// Grab a beta feature form the list to test with
	for k, v := range utilfeature.DefaultMutableFeatureGate.GetAll() {
		if v.PreRelease == featuregate.Beta && v.Default {
			targetFeature = k
			break
		}
	}

	if len(targetFeature) == 0 {
		t.Log("no beta features to test against")
		return
	}

        // see if we iterate the feature map with AllBeta first within a few tries
	for i := 0; i < 10; i++ {
		attemptFunc := func() bool {
			fg := utilfeature.DefaultFeatureGate.DeepCopy()
			originalValue := fg.Enabled(targetFeature)

			func() {
				// Disable all alpha and beta features
				for k, v := range fg.DeepCopy().GetAll() {
					if v.PreRelease == featuregate.Alpha || v.PreRelease == featuregate.Beta {
						defer featuregatetesting.SetFeatureGateDuringTest(t, fg, k, false)()
					}
				}
			}()

			return originalValue != fg.Enabled(targetFeature)
		}

		if attemptFunc() {
			t.Fatal("original value for feature not same as new value after restoration")
			break
		}
	}
}
```

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind bug

#### Which issue(s) this PR fixes:
Fixes https://github.com/kubernetes/kubernetes/issues/116151

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/assign @alexzielenski 
/sig api-machinery testing

### CODE REVIEW

File: staging/src/k8s.io/component-base/featuregate/testing/feature_gate.go [hunk 1]
```diff
@@ -32,6 +32,20 @@ import (
 func SetFeatureGateDuringTest(tb testing.TB, gate featuregate.FeatureGate, f featuregate.Feature, value bool) func() {
 	originalValue := gate.Enabled(f)
 
+	// Specially handle AllAlpha and AllBeta
+	var cleanups []func()
+	if f == "AllAlpha" || f == "AllBeta" {
```
Comment (alexzielenski) [inline staging/src/k8s.io/component-base/featuregate/testing/feature_gate.go:37]: The special features come from map at https://github.com/kubernetes/kubernetes/blob/964529b227331f1f7970aade54123413e795a267/staging/src/k8s.io/component-base/featuregate/feature_gate.go#L61-L64

does it make sense to use those map keys as source of truth for these values? In that case the `cleanup` list would likely need to reset ALL non-special features.

**Review by alexzielenski — APPROVED** [review]
/lgtm

/hold 
in case my idea makes any sense to add

File: staging/src/k8s.io/component-base/featuregate/testing/feature_gate.go (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/component-base/featuregate/testing/feature_gate.go:37]: I don't think I want to export that map or those keys just for this, and I think the cleanup should limit itself to resetting the state of the features that are actually modified by this call

**Comment by liggitt** [comment]
pushed an update to fix (and simplify! yay `git grep) the verify test

**Review by alexzielenski — APPROVED** [review]
/lgtm

**Pull request merged**